content/docs/thinking-in-react.md

```diff
@@ -31,118 +31,118 @@ Our JSON API returns some data that looks like this:
+<p data-height="600" data-theme-id="0" data-slug-hash="BwWzwm" data-default-tab="js" data-user="lacker" data-embed-version="2" class="codepen">参阅 <a href="http://codepen.io">CodePen</a> 上的 <a href="https://codepen.io/gaearon/pen/BwWzwm">React 哲学：第二步</a>。</p>
 <script async src="https://production-assets.codepen.io/assets/embed/ei.js"></script>
 
-Now that you have your component hierarchy, it's time to implement your app. The easiest way is to build a version that takes your data model and renders the UI but has no interactivity. It's best to decouple these processes because building a static version requires a lot of typing and no thinking, and adding interactivity requires a lot of thinking and not a lot of typing. We'll see why.
+现在我们已经确定了组件层级，可以着手编写对应的应用了。此时，最简单的办法是先用已有的数据模型创建一个不包含交互功能的 UI。编写一个应用的静态版本时，往往要进行大量输入，而不需要考虑太多；添加交互功能时，考虑的问题更多，却不怎么需要输入。所以，将这两个过程分开进行更为合适。你会慢慢意识到其中的原因。
 
-To build a static version of your app that renders your data model, you'll want to build components that reuse other components and pass data using *props*. *props* are a way of passing data from parent to child. If you're familiar with the concept of *state*, **don't use state at all** to build this static version. State is reserved only for interactivity, that is, data that changes over time. Since this is a static version of the app, you don't need it.
+在构建应用的静态版本时，一个组件往往会重用其他更小的组件，并通过 *props* 传递所需的数据。*props* 是父组件向子组件传递数据的方式。即使你已经熟悉了 *state* 的概念，也**完全不应该使用 state** 构建静态版本。State 代表了随时间会产生变化的数据，应当仅在实现交互时使用。所以构建应用的静态版本时，你不会用到它。
 
-You can build top-down or bottom-up. That is, you can either start with building the components higher up in the hierarchy (i.e. starting with `FilterableProductTable`) or with the ones lower in it (`ProductRow`). In simpler examples, it's usually easier to go top-down, and on larger projects, it's easier to go bottom-up and write tests as you build.
+你可以自上而下或者自下而上构建应用：自上而下意味着首先编写层级较高的组件（例如 `FilterableProductTable`），自下而上意味着从最基本的组件开始编写（例如 `ProductRow`）。当你的应用比较简单时，使用自上而下的方式更方便；对于较为庞大的项目来说，自下而上地构建，并同时为低层组件编写测试是更加简单的方式。
 
-At the end of this step, you'll have a library of reusable components that render your data model. The components will only have `render()` methods since this is a static version of your app. The component at the top of the hierarchy (`FilterableProductTable`) will take your data model as a prop. If you make a change to your underlying data model and call `ReactDOM.render()` again, the UI will be updated. It's easy to see how your UI is updated and where to make changes since there's nothing complicated going on. React's **one-way data flow** (also called *one-way binding*) keeps everything modular and fast.
+到此为止，你应该已经有了一个可重用的组件库来渲染你的数据模型。由于我们构建的是静态版本，这些组件应该只有唯一的 `render()` 方法。最顶层的组件 `FilterableProductTable` 通过 props 接受你的数据模型。一旦你的数据模型发生了改变，`ReactDOM.render()` 就会被调用，从而更新 UI。所幸这个组件并不复杂，所以很容易修改，也很容易看到 UI 的具体改变。React **单向数据流**（也叫*单向绑定*）的思想使得组件模块化，易于快速开发。
```

Comment:
The components will only have `render()` methods since this is a static version of your app.
这句应该翻译为：由于我们构建的是静态版本，所以这些组件目前只需提供 `render()` 方法来用于渲染。

If you make a change to your underlying data model and call `ReactDOM.render()` again, the UI will be updated.
这句意思是 model 变化、调用 render 函数、UI 相应变化，应该翻译为：如果你的数据模型发生了改变，再次调用 `ReactDOM.render()`，UI 就会相应地被更新。

It's easy to see how your UI is updated and where to make changes since there's nothing complicated going on.
这句承接上句，是说上句描述的基本原理并不复杂，应该翻译为：数据模型变化、调用 render 函数、UI 相应变化，这个过程并不复杂，也很容易看清楚 UI 是如何被更新的，以及是在哪里被更新。